_includes/pbc/all/install-features/202404.0/install-the-comments-feature.md

```diff
@@ -0,0 +1,475 @@
+| CommentWidgetRouteProviderPlugin | SprykerShop\Yves\CommentWidget\Plugin\Router |
+
+**src/Pyz/Yves/Router/RouterDependencyProvider.php**
+
+```php
+<?php
+
+namespace Pyz\Yves\Router;
+
+use Spryker\Yves\Router\RouterDependencyProvider as SprykerRouterDependencyProvider;
+use SprykerShop\Yves\CommentWidget\Plugin\Router\CommentWidgetRouteProviderPlugin;
+
+class RouterDependencyProvider extends SprykerRouterDependencyProvider
+{
+    /**
+     * @return \Spryker\Yves\RouterExtension\Dependency\Plugin\RouteProviderPluginInterface[]
+     */
+    protected function getRouteProvider(): array
+    {
+        return [
+            new CommentWidgetRouteProviderPlugin(),
+        ];
+    }
+}
+```
+
+{% info_block warningBox "Verification" %}
+
+Log in as a customer and open `https://mysprykershop.com/comment/0adafdf4-cb26-477d-850d-b26412fbd382/tag/add?returnUrl=/cart`. Make sure that the error flash message is displayed.
+
+{% endinfo_block %}
+
+### 4) Set up widgets
+
+1. Register the following plugins to enable widgets:
+
+| PLUGIN              | SPECIFICATION      | PREREQUISITES | NAMESPACE                             |
+|---------------------|--------------------|---------------|---------------------------------------|
+| CommentThreadWidget | Displays comments. |               | SprykerShop\Yves\CommentWidget\Widget |
+
+**src/Pyz/Yves/ShopApplication/ShopApplicationDependencyProvider.php**
+
+```php
+<?php
+
+namespace Pyz\Yves\ShopApplication;
+
+use SprykerShop\Yves\CommentWidget\Widget\CommentThreadWidget;
+use SprykerShop\Yves\ShopApplication\ShopApplicationDependencyProvider as SprykerShopApplicationDependencyProvider;
+
+class ShopApplicationDependencyProvider extends SprykerShopApplicationDependencyProvider
+{
+	/**
+	 * @return list<String>
+	 */
+	protected function getGlobalWidgets(): array
+	{
+		return [
+			CommentThreadWidget::class,
+		];
+	}
+}
+```
+
+2. Enable Javascript and CSS changes:
+
+```bash
+console frontend:yves:build
+```
+
+{% info_block warningBox "Verification" %}
+
+Verify the following widgets have been registered by adding the respective code snippets to a Twig template:
+
+| MODULE              | TEST                                                                                                                                                                                         | VERIFICATION |
+|---------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------| - |
+| CommentThreadWidget | `{% raw %}{%{% endraw %} widget 'CommentThreadWidget' args \[...\] only {% raw %}%}{% endraw %}{% raw %}{%{% endraw %} endwidget {% raw %}%}{% endraw %}` | Add a comment on the **Cart** page. Submit the order. Go to the order details page and make sure the comment threat you've added is displayed. |
```

Comment:
Are we verifying just *comments* or *comment threads*?

Reply:
The prerequisite for a comment to appear on the order detail page is to add it to the cart.

Reply:
I understand the prerequisite. Trying to figure out if there are actual comment threads, not just comments. Comment threads like in slack or reddit for example.

Reply:
I would say, it's more like comments without hierarchy.